modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -609,17 +608,14 @@ private CompletableFuture<HybridTimestamp> stopLeaseProlongation(
                         .thenCompose(unused -> {
                             NetworkMessage response = futs.stream()
                                     .map(CompletableFuture::join)
-                                    .filter(Objects::nonNull)
+                                    .filter(resp -> resp instanceof StopLeaseProlongationMessageResponse
+                                            && ((StopLeaseProlongationMessageResponse) resp).deniedLeaseExpirationTime() != null)
```

Comment:
How do we use deniedLeaseExpirationTime()? I mean other cases besides checking whether it's null or not?

Reply:
Is that correct that you expect either all nulls or partially nulls, partially !nulls, but all !nulls will have same value?